src/main/java/pl/psnc/dei/iiif/Converter.java

```diff
@@ -0,0 +1,225 @@
+		cleanDirectory(new File(conversionDirectory, record.getIdentifier()));
+		List<File> filesToConvert = saveFilesInTempDirectory(filesUrls);
+
+		List<String> storedFilesIds;
+//		if(RecordTransferValidationUtil.checkIfTransferPossible(recordJson)) {TODO change after merge with EN-63
+		storedFilesIds = fs.storeFiles(record, filesToConvert);    /** IIIF record case */
+//		} else {
+		storedFilesIds = fs.storeFiles(record, convertFiles(filesToConvert));
+//		}
+
+		cleanDirectory(new File(conversionDirectory, record.getIdentifier()));
+		record.setIiifManifest(getManifest(storedFilesIds).toString());
+	}
+
+	private List<URL> extractFilesUrls(JsonObject aggregatorData) {
+		String mainFileUrl = aggregatorData.get("isShownBy").getAsString().value();
+		String mainFileFormat = mainFileUrl.substring(mainFileUrl.lastIndexOf('.'));
+
+		List<String> links = new ArrayList<>();
+		links.add(mainFileUrl);
+		if (aggregatorData.get("hasView") != null)
+			links.addAll(aggregatorData.get("hasView").getAsArray().stream()
+					.map(e -> e.getAsString().value())
+					.filter(s -> s.endsWith(mainFileFormat))
+					.collect(Collectors.toList()));
+
+		List<URL> result = new ArrayList<>();
+		for (String link : links) {
+			try {
+				result.add(new URL(link));
+			} catch (MalformedURLException e) {
+				logger.error("Incorrect file URL for record: {}, url: {}", record.getIdentifier(), link, e);
+			}
+		}
+
+		return result;
+	}
+
+	private List<File> saveFilesInTempDirectory(List<URL> filesUrls) throws ConversionException {
+		File tempDir = new File(conversionDirectory, record.getIdentifier() + "/src");
+		tempDir.mkdirs();
+
+		for (URL url : filesUrls) {
+			try {
+				String filePath = url.getPath();
+				String fileName = filePath.substring(filePath.lastIndexOf("/"));
+				File tempFile = new File(tempDir, fileName);
+				FileUtils.copyURLToFile(url, tempFile);
+			} catch (IOException e) {
+				logger.error("Couldn't get file: {}", url.toString(), e);
+			}
+		}
+
+		if (tempDir.listFiles().length == 0) {
+			throw new ConversionException("Couldn't get even 1 file, conversion aborted!");
+		} else {
+			return Arrays.asList(tempDir.listFiles());
+		}
+	}
+
+	private List<File> convertFiles(List<File> filesToConvert) throws InterruptedException, IOException {
+		File outDir = new File(conversionDirectory, record.getIdentifier() + "/out");
+		outDir.mkdirs();
+
+		if (filesToConvert.get(0).getAbsolutePath().endsWith("pdf")) {
+			String pdfConversionScript = new ClassPathResource("pdf_to_pyramid_tiff.sh").getFile().getAbsolutePath();
+			File pdfFile = filesToConvert.get(0);
+			try {
+				executor.runCommand(Arrays.asList(
+						pdfConversionScript,
+						pdfFile.getAbsolutePath(),
+						outDir.getAbsolutePath()));
+			} catch (ConversionException e) {
+				logger.error("Conversion failed for file: " + pdfFile.getName() + " from record: " + record.getIdentifier());
+			}
+		} else {
+			for (File file : filesToConvert) {
```

Comment:
All or nothing vs partial results approach. Should we accept that some files cannot be converted?